ISSUE-1214: Add http proxy support as a top level configuration

priyank5485 opened the pull request:
Add http proxy support as a top level configuration

sapinamin commented:
I have submitted the ambari PR https://github.com/hortonworks/hdf_ambari_mp/pull/186 and tested it with jar with this PR. Please merge it.

arunmahadevan commented:
@priyank5485 
The proxy* configs are now changed to httpProxy* . What about users who are currently using the old settings (proxy* setting under streams module) ? 

For non-ambari installs, if they have the old configs, do we throw any exceptions with a message asking them to put the new top level configs ?

For Ambari installs, does ambari take care of automatically copying the old config and putting it under the new top level configs?

sapinamin commented:
@arunmahadevan "For Ambari installs, does ambari take care of automatically copying the old config and putting it under the new top level configs?"
- yes, it does.

arunmahadevan commented on streams/runners/storm/actions/src/main/java/com/hortonworks/streamline/streams/actions/storm/topology/StormTopologyActionsImpl.java:
```diff
@@ -454,13 +454,13 @@ private TopologyLayout copyTopologyLayout(TopologyLayout topology, TopologyDag r
             args.add((String) conf.get("mavenRepoUrl"));
 
-            String proxyUrl = (String) conf.get("proxyUrl");
+            String proxyUrl = (String) conf.get("httpProxyUrl");
```
use the constant here?

priyank5485 replied:
Addressed

arunmahadevan commented on streams/runners/storm/actions/src/main/java/com/hortonworks/streamline/streams/actions/storm/topology/StormTopologyActionsImpl.java:
```diff
@@ -454,13 +454,13 @@ private TopologyLayout copyTopologyLayout(TopologyLayout topology, TopologyDag r
 
-                String proxyUsername = (String) conf.get("proxyUsername");
-                String proxyPassword = (String) conf.get("proxyPassword");
+                String proxyUsername = (String) conf.get("httpProxyUsername");
```
use the constant here?

priyank5485 replied:
Addressed

arunmahadevan commented on streams/runners/storm/actions/src/main/java/com/hortonworks/streamline/streams/actions/storm/topology/StormTopologyActionsImpl.java:
```diff
@@ -454,13 +454,13 @@ private TopologyLayout copyTopologyLayout(TopologyLayout topology, TopologyDag r
-                String proxyUsername = (String) conf.get("proxyUsername");
-                String proxyPassword = (String) conf.get("proxyPassword");
+                String proxyUsername = (String) conf.get("httpProxyUsername");
+                String proxyPassword = (String) conf.get("httpProxyPassword");
```
use the constant here?

priyank5485 replied:
Addressed

priyank5485 commented:
@arunmahadevan I have addressed your comments. For old configs under streams module for non ambari installs I have added a warn level message. Please let me know if it addresses your concern

arunmahadevan commented on webservice/src/main/java/com/hortonworks/streamline/webservice/StreamlineApplication.java:
```diff
@@ -272,6 +272,15 @@ private void registerResources(StreamlineConfiguration configuration, Environmen
             initConfig.put(Constants.CONFIG_AUTHORIZER, authorizer);
             initConfig.put(Constants.CONFIG_SECURITY_CATALOG_SERVICE, securityCatalogService);
             initConfig.put(Constants.CONFIG_SUBJECT, subject);
+            if (configuration.getHttpProxyUrl() == null || configuration.getHttpProxyUrl().isEmpty()) {
```
to catch old configs it should be
```
configuration.getHttpProxyUrl() != null && !configuration.getHttpProxyUrl().isEmpty()
```

priyank5485 replied:
To catch old configs it should should be initConfig.getProperties().get("proxyUrl"); I will change around the logic for detecting and exit in that case.

priyank5485 replied:
I changed my mind and instead of exiting we will use top level configs.

arunmahadevan commented on webservice/src/main/java/com/hortonworks/streamline/webservice/StreamlineApplication.java:
```diff
@@ -272,6 +272,15 @@ private void registerResources(StreamlineConfiguration configuration, Environmen
+                LOG.warn("No http proxy configuration found. If SAM server is behind an http proxy, please move proxyUrl, proxyUsername and " +
+                        "proxyPassword configuration properties under streams module to httpProxyUrl, httpProxyUsername and httpProxyPassword respectively at" +
+                        " top level in your streamline.yaml");
+            }
```
It would be better to either throw an exception and exit or LOG the warning and use these settings if the top level settings are not present.

arunmahadevan commented:
>For old configs under streams module for non ambari installs I have added a warn level message.

We should detect old configs (under storage provider configuration) and probably throw an exception and exit or LOG warning and use these settings if the new top level settings are not set.

priyank5485 commented:
@arunmahadevan Please check the latest commit to see if it addresses your last comment.

arunmahadevan commented:
+1
